studio/src/components/checks/checks-config.tsx

```diff
@@ -0,0 +1,127 @@
+import { useUser } from "@/hooks/use-user";
+import { SubmitHandler, useZodForm } from "@/hooks/use-form";
+import { useMutation } from "@connectrpc/connect-query";
+import { EnumStatusCode } from "@wundergraph/cosmo-connect/dist/common/common_pb";
+import {
+  updateNamespaceChecksConfig,
+} from "@wundergraph/cosmo-connect/dist/platform/v1/platform-PlatformService_connectquery";
+import {
+  GetNamespaceChecksConfigurationResponse,
+} from "@wundergraph/cosmo-connect/dist/platform/v1/platform_pb";
+import { Input } from "@/components/ui/input";
+import { Button } from "../ui/button";
+import {
+  Form,
+  FormControl,
+  FormDescription,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+} from "@/components/ui/form";
+import { useToast } from "../ui/use-toast";
+import { z } from "zod";
+import {useMemo} from "react";
+
+export const ChecksConfig = ({
+  namespace,
+  data,
+}: {
+  namespace: string;
+  data: GetNamespaceChecksConfigurationResponse;
+}) => {
+  const user = useUser();
+
+  const schema = z.object({
+    timeframeInDays: z.coerce.number().min(1).max(data.timeframeLimitInDays),
+  });
+
+  type ChecksConfigurationInput = z.infer<typeof schema>;
+
+  const { mutate, isPending } = useMutation(updateNamespaceChecksConfig);
+
+  const { toast } = useToast();
+
+  const form = useZodForm<ChecksConfigurationInput>({
+    schema,
+    mode: "onChange",
+    values: useMemo(() => ({ timeframeInDays: data.timeframeInDays }), [data]),
+  });
+
+  const onSubmit: SubmitHandler<ChecksConfigurationInput> = (data) => {
+    mutate(
+      {
+        namespace,
+        timeframeInDays: data.timeframeInDays,
+      },
+      {
+        onSuccess: (d) => {
+          if (d.response?.code === EnumStatusCode.OK) {
+            toast({
+              description: "Checks configuration updated successfully.",
+              duration: 3000,
+            });
+          } else if (d.response?.details) {
+            toast({ description: d.response.details, duration: 3000 });
+          }
+        },
+        onError: () => {
+          toast({
+            description:
+              "Could not update the checks configuration. Please try again.",
+            duration: 3000
+          });
+        },
+      }
+    )
+  };
+
+  return (
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit(onSubmit)}
+        className="flex flex-col gap-y-4"
+      >
+        <FormField
+          control={form.control}
+          name="timeframeInDays"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Days to consider</FormLabel>
+              <FormControl>
+                <Input
```

Comment:
To maintain consistency should we go with a dropdown instead of a input?

Reply:
As in a set number of options? Lets say
- 1 day
- 3 days
- 1 week
- 2 weeks

and so on?

Reply:
yep, we can do 
- 3 days
- 7 days
- 14 days
- 30 days ...